tests/test_worlds.py

```diff
@@ -37,15 +36,17 @@ def setUp(self):
             'ComposedShader is experimental.',
             # the ros controller of complete_test.wbt is started when loading the world because the robot-window is open
             'Failed to contact master at',
+            'Cannot initialize the sound engine',
+            'System below the minimal requirements',
             self.crashError  # To remove once #6125 is fixed
         ]
         # Get all the worlds from projects
         self.worlds = []
-        for directory in ['projects']:
-            for rootPath, dirNames, fileNames in os.walk(os.environ['WEBOTS_HOME'] + os.sep + directory):
-                for fileName in fnmatch.filter(fileNames, '*.wbt'):
-                    world = os.path.join(rootPath, fileName)
-                    self.worlds.append(world)
+        root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
I am not sure about that.
In the case of tests in the CI, I think that what we do is:
- Download the artifact that contain the built distrib of webots
- Set the $WEBOTS_HOME to point to the distrib
- Run the tests

However as the distrib does not contain the tests, we run the tests from the cloned github repo.
So the $WEBOTS_HOME does not correspond to `../..`

Reply:
Haven't checked if it's right or wrong, just wanted to point out that contrary to the test_suite, the test_world targets the worlds of the distrib (since they are distributed) as opposed to targetting the worlds from the cloned repo (which we need to do for the test_suite since they aren't included)

Reply:
Until https://github.com/cyberbotics/webots/pull/5087/commits/d1562a65877419af4fda5caa5df94a97d3218a94, I was looking for worlds in the `$WEBOTS_HOME/projects` directory. But this was leading to only 294 worlds to be tested. Then I switched to this method, which is the one used for the current `world-update` test (297 worlds). But if we aim at testing only the distributed worlds (which seem to be 294), it would be more correct to use WEBOTS_HOME to find worlds to test.